src/bin/server/rpc.rs

```diff
@@ -1,19 +1,96 @@
+use std::net::SocketAddr;
+
+use drop::crypto::key::exchange::{self, Exchanger};
+use drop::crypto::sign;
+use drop::net::{ConnectorExt, TcpConnector, TcpListener};
+use drop::system::manager::{Handle, SystemManager};
+use drop::system::sampler::AllSampler;
+use drop::system::sender::NetworkSender;
+use drop::system::System;
+use murmur::MurmurConfig;
+use sieve::{self, Sieve, SieveConfig, SieveMessage};
+
 use tonic::Response;
 
 tonic::include_proto!("at2");
 
 use at2_server::At2;
 pub use at2_server::At2Server;
 
-#[derive(Default)]
-pub struct Service {}
+type M = u64;
+
+pub struct Service {
+    handle: sieve::SieveHandle<M, NetworkSender<SieveMessage<M>>, sieve::Fixed>,
+}
+
+use super::config;
+
+impl Service {
+    pub async fn new(
+        listener_addr: SocketAddr,
+        network_keypair: exchange::KeyPair,
+        sign_keypair: sign::KeyPair,
+        network: Vec<config::Node>,
+    ) -> Self {
+        let exchanger = Exchanger::new(network_keypair);
+
+        let listener = TcpListener::new(listener_addr, exchanger.clone())
+            .await
+            .expect("failed to listen for incoming connections");
+
+        let (addrs, keys): (Vec<_>, Vec<_>) = network
+            .iter()
+            .map(|node| (node.address, node.public_key))
+            .unzip();
+        let connector = TcpConnector::new(exchanger).retry();
+        let mut system = System::new_with_connector(&connector, keys, addrs).await;
+
+        let _ = system.add_listener(listener).await;
+        let manager = SystemManager::new(system);
+
+        let sieve = Sieve::new(
+            sign_keypair,
+            sieve::Fixed::new_local(),
+            SieveConfig {
+                sieve_sample_size: 1,
+                murmur: MurmurConfig {
+                    murmur_gossip_size: network.len(),
+                    ..Default::default()
+                },
+                ..Default::default()
+            },
+        );
+
+        let sampler = AllSampler::default();
+
+        Self {
+            handle: manager
+                .run(sieve, sampler, num_cpus::get())
+                .await
+                .processor_handle(),
```

Comment:
You may want to keep the `SystemHandle` around to add connections to the system later and be aware of errors

Reply:
> to add connections to the system later

Hum, I don't think we need to support adding new connections, the network is fixed.

> and be aware of errors

Right, is there is some errors aimed at the user directly (such as a failed broadcast)? Or are theses for the node maintainer?

Reply:
> Hum, I don't think we need to support adding new connections, the network is fixed.

You do. Disonnects are one case where it is needed to re-add the connection to the system.

> Right, is there is some errors aimed at the user directly (such as a failed broadcast)? Or are theses for the node maintainer?

Most errors from the `SystemHandle` are aimed at the local node manager. User related errors are coming mostly from the `Handle::broadcast` function

Reply:
> Of course all of this can be done later and is not required for early testing

Noted, I'm adding some TODOS :)